From 68bcc74a6347c314d95b61240a1849868d021214 Mon Sep 17 00:00:00 2001
From: Mykola Lysenko <Mykola.Lysenko@amd.com>
Date: Mon, 14 Dec 2015 19:08:47 +0800
Subject: [PATCH 0615/1110] drm/amd/dal: validate MST connector modes

Same validation should be applied for MST connector,
as for regular one.

In this change default validation function is exposed
globally and used for MST connector

Signed-off-by: Mykola Lysenko <Mykola.Lysenko@amd.com>
Signed-off-by: Harry Wentland <harry.wentland@amd.com>
Acked-by: Harry Wentland<harry.wentland@amd.com>
---
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c | 9 +--------
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c     | 5 +++--
 drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.h     | 4 ++++
 3 files changed, 8 insertions(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
index f2926ff..fb71d88 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_mst_types.c
@@ -208,13 +208,6 @@ static int dm_dp_mst_get_modes(struct drm_connector *connector)
 	return ret;
 }
 
-static enum drm_mode_status
-dm_dp_mst_mode_valid(struct drm_connector *connector,
-			struct drm_display_mode *mode)
-{
-	return MODE_OK;
-}
-
 static struct drm_encoder *dm_mst_best_encoder(struct drm_connector *connector)
 {
 	struct amdgpu_connector *amdgpu_connector = to_amdgpu_connector(connector);
@@ -224,7 +217,7 @@ static struct drm_encoder *dm_mst_best_encoder(struct drm_connector *connector)
 
 static const struct drm_connector_helper_funcs dm_dp_mst_connector_helper_funcs = {
 	.get_modes = dm_dp_mst_get_modes,
-	.mode_valid = dm_dp_mst_mode_valid,
+	.mode_valid = amdgpu_dm_connector_mode_valid,
 	.best_encoder = dm_mst_best_encoder,
 };
 
diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
index 633f0de..fff17cd 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.c
@@ -1221,7 +1221,8 @@ static int get_modes(struct drm_connector *connector)
 	return amdgpu_connector->num_modes;
 }
 
-static int mode_valid(struct drm_connector *connector,
+int amdgpu_dm_connector_mode_valid(
+		struct drm_connector *connector,
 		struct drm_display_mode *mode)
 {
 	int result = MODE_ERROR;
@@ -1286,7 +1287,7 @@ amdgpu_dm_connector_helper_funcs = {
 	* in get_modes call back, not just return the modes count
 	*/
 	.get_modes = get_modes,
-	.mode_valid = mode_valid,
+	.mode_valid = amdgpu_dm_connector_mode_valid,
 	.best_encoder = best_encoder
 };
 
diff --git a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.h b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.h
index 1b8b3eb..5d1152e 100644
--- a/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.h
+++ b/drivers/gpu/drm/amd/dal/amdgpu_dm/amdgpu_dm_types.h
@@ -92,6 +92,10 @@ void amdgpu_dm_connector_init_helper(
 	const struct dc_link *link,
 	int link_index);
 
+int amdgpu_dm_connector_mode_valid(
+	struct drm_connector *connector,
+	struct drm_display_mode *mode);
+
 extern const struct drm_encoder_helper_funcs amdgpu_dm_encoder_helper_funcs;
 
 #endif		/* __AMDGPU_DM_TYPES_H__ */
-- 
2.7.4

